src/back/project/src/api/simple_crud.hpp

```diff
@@ -0,0 +1,132 @@
+#pragma once
+
+#include <shared/errors_catchit.hpp>
+#include <shared/parse/request.hpp>
+#include <shared/schemas.hpp>
+#include <shared/type_utils.hpp>
+
+#include <map>
+
+#include <type_traits>
+#include <userver/formats/json/value.hpp>
+#include <userver/components/component_config.hpp>
+#include <userver/components/component_context.hpp>
+#include <userver/server/handlers/http_handler_json_base.hpp>
+#include <userver/utest/using_namespace_userver.hpp>
+#include "userver/utils/boost_uuid4.hpp"
+
+namespace svetit::project::handlers {
+
+template<typename Service, typename T, const char* handlerName>
+class SimpleCrud : public server::handlers::HttpHandlerJsonBase {
+	using m = server::http::HttpMethod;
+public:
+	static constexpr std::string_view kName = handlerName;
+
+	explicit SimpleCrud(
+		const components::ComponentConfig& conf,
+		const components::ComponentContext& ctx)
+		: server::handlers::HttpHandlerJsonBase{conf, ctx}
+		, _s{ctx.FindComponent<Service>()}
+		, _mapHttpMethodToSchema{LoadSchemas(kName, _s.GetJSONSchemasPath())} {}
+
+	formats::json::Value HandleRequestJsonThrow(
+		const server::http::HttpRequest& req,
+		const formats::json::Value& body,
+		server::request::RequestContext&) const override
+	{
+		try {
+			// TODO: Add headers to API
+			// - X-User
+			// - X-Space-Id
+			// - X-Space-Role
+			const auto params = ValidateRequest(_mapHttpMethodToSchema, req, body);
+
```

Comment:
Нужно достать эти параметры и передавать их во все функции.
userId нужно добавлять в логирование там где оно используется.
spaceId нужно передавать во все sql запросы.
```suggestion
			const auto userId = params["X-User"].As<std::string>();
			const auto spaceId = params["X-Space-Id"].As<std::string>();

```

Reply:
По spaceId - готово, уже передаётся во все sql запросы. 

по userId - логирование, если не ошибаюсь, в мс Проекты сейчас происходит только вот тут https://github.com/svetit-io/svetit/blob/feature/100-endpoint-validation/src/back/deps/svetit/shared/schemas.cpp#L105 (работа со схемами) и вот тут https://github.com/svetit-io/svetit/blob/feature/100-endpoint-validation/src/back/deps/svetit/shared/errors_catchit.cpp#L21 (стандартный верхнеуровневый обработчик исключений). Нужно ли передавать userId в эти места?

Reply:
Добавил